### DISCUSSION TITLE
Setting a list of prohibited URLs in Sandboxie.ini

### DISCUSSION THREAD

**Discussion by wafu77** [sandboxie-plus/Sandboxie#2715] [Documents and notes]
Is it possible to prohibit connections to a URL by setting "Sandboxie.ini"?
We have confirmed that we can prohibit connections by IP address, but we do not know how to set it by URL.

I would like to create a sandbox that cannot access this URL https://ssl.bing.com/profile/history.

**Comment by offhub** [comment]
You cannot block directly with Sandboxie, but you can block using policies.

Example for Edge/Chrome/Firefox/Brave:

- To sandbox config:
> This command will import the following registry file into the sandboxed registry when the sandbox is first initialized.
```
AutoExec=regedit /s "X:\ImportPolicies.reg"
```

- ImportPolicies.reg:
```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\software\Policies\Microsoft\Edge\URLBlocklist]
"1"="https://ssl.bing.com/profile/history"

[HKEY_CURRENT_USER\software\Policies\Google\Chrome\URLBlocklist]
"1"="https://ssl.bing.com/profile/history"

[HKEY_CURRENT_USER\software\Policies\Mozilla\Firefox\WebsiteFilter\Block]
"1"="https://ssl.bing.com/profile/history"

[HKEY_CURRENT_USER\software\Policies\BraveSoftware\Brave\URLBlocklist]
"1"="https://ssl.bing.com/profile/history"

```

**Reply by wafu77** [reply]
Thanks offhub.
I am trying your advice.
I have placed "ImportPolicies.reg" in "C:\TMP\" and run it, but it is still not working.

I ran the command at the command prompt to confirm.
And confirmed that it is added to the registry.

`c:\> regedit /s "C:\TMP\ImportPolicies.reg"`


I will look into the relationship between Sandboxie and the registry.

I have a point of concern.

　Does this setting not affect other sandboxes?
I do not want to restrict this URL in other sandboxes.

**Reply by offhub** [reply]
> I ran the command at the command prompt to confirm.

From sandboxed command line or unsandboxed? (Use sandboxed commandline to test it)
Also if you are running it while the browser is open, it will not be effective, so you need to restart the browser.

> Does this setting not affect other sandboxes?

It will only take effect in sandboxes where the AutoExec command has been added.

**Note**: Since the REG file will be needed every time the sandbox is initialized, you should store it in a permanent location.


https://user-images.githubusercontent.com/6871698/221355230-d182338b-e871-4769-8490-89c60843c95b.mp4

**Reply by wafu77** [reply]
Thanks so much to you, offhub!

Your advice and detailed video helped me to achieve the operation.

The usage you taught me can be used to realize other behaviors.

It helped me a lot.

--
wafu
